### PR TITLE
Update Naming to Latest Networking Spec

### PR DESCRIPTION
**Pull request by nisdas**
- [x] Change to `Status` from `Hello`
- [x] Change to `beacon_blocks_by_root` from `recent_beacon_block`
- [x] Change to `beacon_blocks_by_range` from `beacon_block`
- [x] Remove GoodBye and recent_beacon_blocks messages as they 
are represented as a single ssz encoded field now.
- [x] Fix all tests

### CODE REVIEW

File: beacon-chain/sync/rpc.go [hunk 1]
```diff
@@ -76,3 +77,16 @@ func (r *RegularSync) registerRPC(topic string, base proto.Message, handle rpcHa
 		}
 	})
 }
+
+func copyValues(msg interface{}) interface{} {
+	switch t := msg.(type) {
+	case proto.Message:
+		return proto.Clone(t)
+	case uint64:
```
Comment (prestonvanloon) [inline beacon-chain/sync/rpc.go]: Do you need these cases? Wouldn’t it work with the default/default option?

File: beacon-chain/sync/rpc_goodbye.go [hunk 2]
```diff
@@ -5,18 +5,31 @@ import (
...
-	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
 )
 
+var goodByes = map[uint64]string{
+	0: "Client Shut Down",
+	1: "Irrelevant Network",
+	2: "Fault/Error",
+}
+
+func (r *RegularSync) transformGoodbyeVal(num uint64) string {
```
Comment (prestonvanloon) [inline beacon-chain/sync/rpc_goodbye.go]: This doesn’t need to be a pointer receiver

File: beacon-chain/sync/rpc_goodbye.go [hunk 3]
```diff
@@ -5,18 +5,31 @@ import (
...
+	reason, ok := goodByes[num]
+	if ok {
+		return reason
+	}
+	return "Unknown Goodbye Value Received"
```
Comment (prestonvanloon) [inline beacon-chain/sync/rpc_goodbye.go]: Be sure to include the integer value here

File: beacon-chain/sync/rpc_goodbye.go (hunk 2 above)
Comment (prestonvanloon) [reply beacon-chain/sync/rpc_goodbye.go]: Can we also call this goodbyeMessage?

File: beacon-chain/sync/rpc.go (hunk 1 above)
Comment (nisdas) [reply beacon-chain/sync/rpc.go]: this was a workaround to get the tests to pass, once #3512 is merged in I will just remove these cases and use that instead

File: beacon-chain/sync/rpc.go [hunk 4]
```diff
@@ -25,29 +26,29 @@ type rpcHandler func(context.Context, interface{}, libp2pcore.Stream) error
...
-		&pb.Hello{},
-		r.helloRPCHandler,
+		"/eth2/beacon_chain/req/status/1",
+		&pb.Status{},
+		r.statusRPCHandler,
 	)
 	r.registerRPC(
 		"/eth2/beacon_chain/req/goodbye/1",
-		&pb.Goodbye{},
+		new(uint64),
```
Comment (rauljordan) [inline beacon-chain/sync/rpc.go]: We shouldn't need to do this anymore, just pass in uint64

Comment (nisdas) [reply beacon-chain/sync/rpc.go]: That wont work, uint64 alone is not regarded as a valid interface{}, you have to make it to an expression to make it to pass

**Pull request merged**